actionview/lib/action_view/rendering.rb

```diff
@@ -92,12 +92,16 @@ def rendered_format
       # Find and render a template based on the options given.
       # :api: private
       def _render_template(options) #:nodoc:
-        variant = options[:variant]
+        variant = options.delete(:variant)
```

Comment:
There is any need to mutate now. If so let's keep it. If not we should not
mutate.

On Tue, Jan 20, 2015, 17:37 Ravil Bayramgalin <notifications@github.com>
wrote:

> In actionview/lib/action_view/rendering.rb
> <https://github.com/rails/rails/pull/18546#discussion-diff-23248812>:
>
> > @@ -92,12 +92,16 @@ def rendered_format
> >        # Find and render a template based on the options given.
> >        # :api: private
> >        def _render_template(options) #:nodoc:
> > -        variant = options[:variant]
> > +        variant = options.delete(:variant)
>
> _render_template is not a user-accessible method and methods atop mutate
> options freely (assigning and deleting keys), e. g.
>
>
> https://github.com/rails/rails/blob/master/actionpack/lib/action_controller/metal/rendering.rb#L56
>
> https://github.com/rails/rails/blob/master/actionpack/lib/action_controller/metal/rendering.rb#L63
>
> https://github.com/rails/rails/blob/master/actionpack/lib/abstract_controller/rendering.rb#L107
>
> If we'd like to preserve original arguments unchanged then a duplicate of
> hash is needed at
> https://github.com/brainopia/rails/blob/action_view_render/actionview/lib/action_view/rendering.rb#L122
> or all the methods mutating options should be fixed.
>
> I'm fine with either, tell me what do you prefer and I'll do it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rails/rails/pull/18546/files#r23248812>.
>